tests/ArrayKeysTest.php

```diff
@@ -183,6 +183,26 @@ public function test(bool $v): array {
                     //$b->e["b"] = 123;
                     ',
             ],
+            'intStringKeyAsInt' => [
+                'code' => '<?php
+                    $a = ["15" => "a"];
+                    $b = ["15.7" => "a"];
+                    // since PHP 8 this is_numeric but will not be int key
+                    $c = ["15 " => "a"];
+                    $d = ["-15" => "a"];
+                    // see https://github.com/php/php-src/issues/9029#issuecomment-1186226676
+                    $e = ["+15" => "a"];
+                    $f = ["015" => "a"];
```

Comment:
I'd also add another example with the scientific format - e.g. `1e2`. It's also numeric, but is not converted to int.